Aggregation function calls lead to nil pointer dereference

### Description

When calling aggregation functions like `sum` or `count` it lead to a null pointer dereference error.

### How to replicate

Start the repl at `examples/memkv` and type the following query
`select sum(value) where key ^= '' ;`

The process will panic with the following message
```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x100d714cc]

goroutine 1 [running]:
github.com/c4pt0r/kvql.(*ExecuteCtx).AdjustChunkCache(...)
```

--- a/aggregate_plan.go
+++ b/aggregate_plan.go
@@ -321,7 +321,7 @@ func (a *AggregatePlan) Batch(ctx *ExecuteCtx) ([][]Column, error) {
 		}
 	}
 	if a.Limit < 0 {
-		return a.batch(nil)
+		return a.batch(ctx)
 	}
 	var (
 		rows   [][]Column
@@ -332,7 +332,7 @@ func (a *AggregatePlan) Batch(ctx *ExecuteCtx) ([][]Column, error) {
 	)
 	for a.skips < a.Start {
 		restSkips := a.Start - a.skips
-		rows, err = a.batch(nil)
+		rows, err = a.batch(ctx)
 		if err != nil {
 			return nil, err
 		}
@@ -363,7 +363,7 @@ func (a *AggregatePlan) Batch(ctx *ExecuteCtx) ([][]Column, error) {
 		return ret, nil
 	}
 	for !finish {
-		rows, err = a.batch(nil)
+		rows, err = a.batch(ctx)
 		if err != nil {
 			return nil, err
 		}

--- a/aggregate_plan.go
+++ b/aggregate_plan.go
@@ -315,7 +315,7 @@ func (a *AggregatePlan) createAggrRow(kvp KVPair, ctx *ExecuteCtx) ([]*AggrPlanF
 
 func (a *AggregatePlan) Batch(ctx *ExecuteCtx) ([][]Column, error) {
 	if !a.prepared {
-		err := a.prepareBatch(nil)
+		err := a.prepareBatch(ctx)
 		if err != nil {
 			return nil, err
 		}

